jaydebeapi/__init__.py

```diff
@@ -484,19 +484,24 @@ def _close_last(self):
     # but I'm not sure when __del__ will be called
     __del__ = _close_last
 
-    def _set_stmt_parms(self, prep_stmt, parameters):
+    def _set_stmt_parms(self, prep_stmt, parameters, extra_args):
         for i in range(len(parameters)):
-            # print (i, parameters[i], type(parameters[i]))
             prep_stmt.setObject(i + 1, parameters[i])
+        
+        query_timeout = extra_args.pop("query_timeout", None)
 
-    def execute(self, operation, parameters=None):
+        # https://docs.oracle.com/javase/7/docs/api/java/sql/Statement.html#setQueryTimeout(int)
+        if query_timeout is not None:
+            prep_stmt.setQueryTimeout(int(query_timeout))  # unit: seconds
+
+    def execute(self, operation, parameters=None, **kwargs):
```

Comment:
I would prefer adding the querytimeout parameter to the .connect() method. This would keep method calls compatible to python db api.